src/sensors/icm20948sensor.cpp

```diff
@@ -396,6 +410,28 @@ void ICM20948Sensor::motionLoop() {
                     lastData = millis();
                 }
             }
+            if (sendAcceleration)
+            {
+                acceleration[0] = (float)dmpData.Raw_Accel.Data.X;
+                acceleration[1] = (float)dmpData.Raw_Accel.Data.Y;
+                acceleration[2] = (float)dmpData.Raw_Accel.Data.Z;
+
+                // get the component of the acceleration that is gravity
+                float gravity[3];
+                gravity[0] = 2 * (quaternion.x*quaternion.z - quaternion.w*quaternion.y);
+                gravity[1] = 2 * (quaternion.w*quaternion.x + quaternion.y*quaternion.z);
+                gravity[2] = quaternion.w*quaternion.w - quaternion.x*quaternion.x - quaternion.y*quaternion.y + quaternion.z*quaternion.z;
```

Comment:
Isn't `gravity` a field of `Sensor`? Why are you not using `this->` here but everywhere else?

Reply:
gravity is not a field of Sensor it is defined a few lines before. Would you like it to be a field?